### ISSUE TITLE
Provide bindings for different languages

### ISSUE THREAD

**Issue by XVilka** [AgentD/squashfs-tools-ng#16]
Would simplify life for a lot of different projects

**Comment by AgentD** [comment]
Yes, moving a sensible chunk of the support libraries to a shared library with a clean API and a slightly more permissive license that allows dynamic linking sounds worthwhile. I will look into that.

**Comment by AgentD** [comment]
I released version 0.7 today (TODO: add link to the announcement once the mail gets through gray-listing).

The new version builds a library `libsquashfs` that can be used by 3rd party applications to do all kinds of things with SquashFS images, also allowing low level poking in the data structures or embedding SquashFS in a custom container format (or vice versa; SquashFS easily allows adding custom extra data while still staying compatible).

TODOs from here on:
- Language bindings (IMO highest priority for now: Python)
- Example programs
